### DISCUSSION TITLE
How can "com.apple.pkg.CLTools_Executables" be missing, yet Command Line Tools still installed?

### DISCUSSION THREAD

**Discussion by ugultopu** [Homebrew/discussions#799] [Everyday usage]
This is more of a Apple systems question rather than Homebrew. On my installation, there is no package "com.apple.pkg.CLTools_Executables" (or "com.apple.pkg.CLTools_Base"):

    $ pkgutil --pkgs | grep ^com.apple
    com.apple.update.fullbundleupdate.19H2
    com.apple.pkg.update.os.10.15.7Supplemental.19H15
    com.apple.pkg.EmbeddedOSFirmware
    com.apple.update.fullbundleupdate.19H15
    com.apple.update.fullbundleupdate.19G2021
    com.apple.pkg.update.os.10.15.7.19H2
    com.apple.pkg.SecureBoot
    com.apple.pkg.update.os.SecUpd2020-001Catalina.19H114
    com.apple.pkg.update.os.10.15.6Supplemental.19G2021
    com.apple.pkg.Core
    com.apple.update.fullbundleupdate.19H114
    com.apple.pkg.XProtectPlistConfigData_10_15.16U4147
    com.apple.files.data-template
    com.apple.pkg.XProtectPlistConfigData_10_15.16U4145
    com.apple.pkg.MRTConfigData_10_15.16U4142
    com.apple.pkg.MRTConfigData_10_15.16U4146
    com.apple.pkg.IncompatibleAppList.10_15.16U1923
    com.apple.pkg.Pages10
    com.apple.pkg.iMovie_AppStore
    com.apple.pkg.Keynote10
    com.apple.pkg.Numbers10
    $ ls /Applications/Xcode.app
    ls: /Applications/Xcode.app: No such file or directory

Yet the Command Line Tools still exist:

    $ type git
    git is hashed (/usr/bin/git)
    $ /usr/bin/git --version
    git version 2.24.3 (Apple Git-128)

Also:

    $ brew config
    HOMEBREW_VERSION: 3.0.1
    ORIGIN: https://github.com/Homebrew/brew
    HEAD: c951be8d3c7b339c7e759b7c40aec859e09a70a5
    Last commit: 3 days ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: c941ccdf270bfc57fa3f8c0a14e289abc10cb8bb
    Core tap last commit: 27 hours ago
    Core tap branch: master
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_CASK_OPTS: []
    HOMEBREW_MAKE_JOBS: 4
    Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
    CPU: quad-core 64-bit broadwell
    Clang: 12.0 build 1200
    Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
    Curl: 7.64.1 => /usr/bin/curl
    macOS: 10.15.7-x86_64
    CLT: 12.0.32.28
    Xcode: N/A

At first, I was confused how can Homebrew tell me the CLT version, since neither of the packages are available. Then I realized that it is using [`detect_version_from_clang_version`](https://github.com/Homebrew/brew/blob/3.0.1/Library/Homebrew/os/mac/xcode.rb#L351).

So, how come I have CLT installed (as evident by the outputs of `/usr/bin/git --version` and `brew config`), yet not have the "com.apple.pkg.CLTools_Executables" (or "com.apple.pkg.CLTools_Base") package?

**Comment by SMillerDev** [comment]
I don't think the fact that you have git indicates a presence of the commandline tools. `xcode-select -p` might be clearer. In addition to that, what does `pkgutil --pkg-info=com.apple.pkg.CLTools_Executables` say?

**Reply by ugultopu** [reply]
$ xcode-select -p
    /Library/Developer/CommandLineTools
    $ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
    No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

I thought it does because it says "Apple Git". I thought Git that is installed from Homebrew does not say it.

**Reply by SMillerDev** [reply]
> I thought it does because it says "Apple Git". I thought Git that is installed from Homebrew does not say it.

It just means that you have a git version from Apple. Just like everything else in macOS is from Apple.
You can check the content of the directory `xcode-select -p` returned to see if anything is still installed.

**Reply by ugultopu** [reply]
Yep, `/Library/Developer/CommandLineTools` is not empty.
